bundles/core/src/main/java/com/adobe/cq/commerce/core/components/storefrontcontext/ProductStorefrontContext.java

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ *
+ *    Copyright 2021 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.storefrontcontext;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public interface ProductStorefrontContext extends CommerceStorefrontContext {
+
+    @JsonProperty("productId")
+    Integer getId();
```

Comment:
I assume id being an Integer is given by the events format? Can we make it String also to be future ready with UID and other ID formats?

Reply:
It has to be an integer according to the Event SDK schema: https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/product.ts#L7.